src/dstack/_internal/core/models/profiles.py

```diff
@@ -109,6 +109,10 @@ class ProfileParams(CoreModel):
             description="The cloud-specific instance types to consider for provisioning (e.g., `[p3.8xlarge, n1-standard-4]`)"
         ),
     ]
+    reservation: Annotated[
+        Optional[str],
+        Field(description="The existing reservation for the instances"),
+    ] = None
```

Comment:
I've addressed backward compatibility in [2b93408](https://github.com/dstackai/dstack/pull/1977/commits/2b93408dd4291071c89c8115f0fea639e402fddd), but it would sure be great to additionally validate it. Tested myself against server version, i.e. `0.18.25`.